x-pack/plugins/upgrade_assistant/public/application/app_context.tsx

```diff
@@ -55,3 +27,7 @@ export const useAppContext = () => {
   }
   return ctx;
 };
+
+export const useServices = () => useAppContext().services;
+
+export const usePlugins = () => useAppContext().plugins;
```

Comment:
Minor nit -- it seems strange to me to provide these methods on the interface of this module. Why provide multiple ways of accessing the same thing, especially given how trivial it is to access `services` and `plugins`? These methods aren't doing much work, so I don't think they justify the complexity they add to the interface.

Reply:
I don't mind removing it in favour of just using `useAppContext`, we basically only saving one destruct operation with this. I saw it implemented this way in `snapshot_restore` and `index_management` and figured might as well do the same. I'll patch it up soon 🔧

Reply:
Done with https://github.com/elastic/kibana/pull/109801/commits/c9523a6259c2a2de7af85412c0981e8dc2c028e5